src/dos/drive_virtual.cpp

```diff
@@ -562,19 +544,11 @@ bool Virtual_Drive::GetFileAttr(char *name, uint16_t *attr)
 		*attr = DOS_ATTR_DIRECTORY;
 		return true;
 	}
-	auto cur_file = first_file;
-	while (cur_file) {
-		unsigned int onpos = cur_file->onpos;
-		if (strcasecmp(name, (std::string(onpos ? vfilenames[onpos].shortname +
-		                                                  std::string(1, '\\')
-		                                        : "") +
-		                      cur_file->name)
-		                             .c_str()) == 0) {
-			*attr = (int)(cur_file->isdir ? DOS_ATTR_DIRECTORY // Maybe
-			                              : DOS_ATTR_ARCHIVE); // Read-only?
-			return true;
-		}
-		cur_file = cur_file->next;
+	auto cur_file = find_file_by_name(name);
+	if (cur_file) {
+		*attr = (int)(cur_file->isdir ? DOS_ATTR_DIRECTORY // Maybe
+		                              : DOS_ATTR_ARCHIVE); // Read-only?
+		return true;
```

Comment:
With the manual linked-list (first file, to cur_file, to next link) now abstracted away, the `cur_file` name loses its context. 

Suggest replacing `cur_file` with `vfile = ...`.

Reply:
i will rename all the occurence